catalog/app/containers/Bucket/PackageUpdateDialog.js

```diff
@@ -0,0 +1,933 @@
+  dropMsgErr: {
+    color: t.palette.error.main,
+  },
+  dropMsgWarn: {
+    color: t.palette.warning.dark,
+  },
+  filesContainer: {
+    borderBottom: `1px solid ${t.palette.action.disabled}`,
+    maxHeight: 200,
+    overflowX: 'hidden',
+    overflowY: 'auto',
+  },
+  filesContainerErr: {
+    borderColor: t.palette.error.main,
+  },
+  filesContainerWarn: {
+    borderColor: t.palette.warning.dark,
+  },
+  added: {},
+  modified: {},
+  deleted: {},
+  unchanged: {},
+  fileEntry: {
+    alignItems: 'center',
+    background: t.palette.background.paper,
+    display: 'flex',
+    '&:not(:last-child)': {
+      borderBottomStyle: 'solid',
+      borderBottomWidth: '1px',
+      borderColor: 'inherit',
+    },
+    '&$added': {
+      background: M.colors.green[100],
+    },
+    '&$modified': {
+      background: M.colors.yellow[100],
+    },
+    '&$deleted': {
+      background: M.colors.red[100],
+    },
+  },
+  filePath: {
+    ...t.typography.body2,
+    flexGrow: 1,
+    marginRight: t.spacing(1),
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+    whiteSpace: 'nowrap',
+  },
+  fileSize: {
+    ...t.typography.body2,
+    color: t.palette.text.secondary,
+    marginRight: t.spacing(0.5),
+  },
+  lock: {
+    alignItems: 'center',
+    background: 'rgba(255,255,255,0.9)',
+    border: `1px solid ${t.palette.action.disabled}`,
+    borderRadius: t.shape.borderRadius,
+    bottom: 0,
+    cursor: 'not-allowed',
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'center',
+    left: 0,
+    position: 'absolute',
+    right: 0,
+    top: 0,
+  },
+  progressContainer: {
+    display: 'flex',
+    position: 'relative',
+  },
+  progressPercent: {
+    ...t.typography.h5,
+    alignItems: 'center',
+    bottom: 0,
+    display: 'flex',
+    justifyContent: 'center',
+    left: 0,
+    position: 'absolute',
+    right: 0,
+    top: 0,
+  },
+  progressSize: {
+    ...t.typography.body2,
+    color: t.palette.text.secondary,
+    marginTop: t.spacing(1),
+  },
+}))
+
+function FilesInput({ input, meta, uploads, setUploads, errors = {} }) {
+  const classes = useFilesInputStyles()
+
+  const value = input.value || { added: {}, deleted: {}, existing: {} }
+  const disabled = meta.submitting || meta.submitSucceeded
+  const error = meta.submitFailed && meta.error
+
+  const onInputChange = input.onChange
+
+  const totalSize = React.useMemo(
+    () => Object.values(value.added).reduce((sum, f) => sum + f.size, 0),
+    [value.added],
+  )
+
+  const warn = totalSize > PD.MAX_SIZE
+
+  // eslint-disable-next-line no-nested-ternary
+  const label = error
+    ? errors[error] || error
+    : warn
+    ? `Total size of new files exceeds recommended maximum of ${readableBytes(
+        PD.MAX_SIZE,
+      )}`
+    : 'Drop files here or click to browse'
+
+  const pipeValue = useMemoEq([onInputChange, value], () => (...fns) =>
+    R.pipe(...fns, onInputChange)(value),
+  )
+
+  const onDrop = React.useCallback(
```

Comment:
Seems like files droparea should be standalone component reusing — but not re-implementing — by PackageDialog and PackagUpdateDialog

Reply:
yep, i thought about this, but it will require some parameterization to handle plain file list vs diff-like file list, and i dont think i want to make this refactoring now